Sources/MobileSdk/MDocHolderBLECentral.swift

```diff
@@ -17,20 +19,45 @@ enum DataError: Error {
     case unknownDataTransferPrefix(byte: UInt8)
 }
 
+/// The MDoc holder as a BLE central.
 class MDocHolderBLECentral: NSObject {
+    enum MachineState {
+        case initial, hardwareOn, fatalError, complete, halted
+        case awaitPeripheralDiscovery, peripheralDiscovered, checkPeripheral
+        case awaitRequest, requestReceived, sendingResponse
+        case l2capAwaitRequest, l2capRequestReceived, l2capSendingResponse
+    }
+
     var centralManager: CBCentralManager!
     var serviceUuid: CBUUID
     var callback: MDocBLEDelegate
     var peripheral: CBPeripheral?
+
     var writeCharacteristic: CBCharacteristic?
     var readCharacteristic: CBCharacteristic?
     var stateCharacteristic: CBCharacteristic?
+    var l2capCharacteristic:  CBCharacteristic?
+
     var maximumCharacteristicSize: Int?
     var writingQueueTotalChunks = 0
     var writingQueueChunkIndex = 0
     var writingQueue: IndexingIterator<ChunksOfCountCollection<Data>>?
 
     var incomingMessageBuffer = Data()
+    var outgoingMessageBuffer = Data()
+
+    private var channelPSM: UInt16? = nil
```

Comment:
What's the need for using `UInt16` instead of `Data` or `ChunksOfCountCollection`?